lib/extract-svg-metadata.js

```diff
@@ -0,0 +1,106 @@
+const createConfig = require('svgo/lib/svgo/config');
+const convertSVG = require('svgo/lib/svgo/svg2js');
+const applyPlugins = require('svgo/lib/svgo/plugins');
+const computePathBounds = require('svg-boundings').path;
+
+const extractMapboxIconMetadata = {
```

Comment:
We should consider add more test fixtures to identify the limit of which kinds of SVGs this code will handle, and consider throwing errors in cases where we get SVGs that we can't handle well (nested transforms, groups, different element types). I can forsee difficult to debug cases cropping up in the future if we don't get on top of this.